### PR TITLE
Improve errors when attempting to remove a domain not on the list

### LINKED ISSUE #159: Removal form displays "Error: Domain protected" for subdomains of bulk-preloaded domains
The "Error: Domain protected" message shouldn't appear for subdomains of a bulk-preloaded domain in the removal form, and instead a separate error should appear indicating that the preloaded domain (and not a subdomain) should be entered into the removal form.

### PR DESCRIPTION
**Pull request by nharper**
If a user enters a domain not on the preload list into the removal form,
but it is a subdomain of a preloaded domain, the error message will now
indicate that the entered domain name is a subdomain of a preloaded
domain instead of stating that the entered domain is protected from removal.

Fixes #159

### CODE REVIEW

**Review by ericroman920 — APPROVED** [review]

File: api/api_test.go [hunk 1]
```diff
@@ -33,6 +33,18 @@ var issuesRemovableProtected = hstspreload.Issues{
 	},
 }
 
+var issuesRemovableSubdomain = hstspreload.Issues{
+	Errors: []hstspreload.Issue{
+		{Code: "server.removable.subdomain"},
```
Comment (lgarron) [inline api/api_test.go:38]: Nit: we use `server.remove.*` already.
```suggestion
		{Code: "server.remove.subdomain"},
```

File: api/api_test.go [hunk 2]
```diff
@@ -33,6 +33,18 @@ var issuesRemovableProtected = hstspreload.Issues{
...
+	},
+}
+
+var issuesRemovablePreloadedTld = hstspreload.Issues{
+	Errors: []hstspreload.Issue{
+		{Code: "server.removable.preloaded_tld"},
```
Comment (lgarron) [inline api/api_test.go:44]: Nit: we use `server.remove.*` already.
```suggestion
		{Code: "server.remove.preloaded_tld"},
```

**Review by lgarron — APPROVED** [review]
I have some optional nits about error code consistency, but these messages will be very useful!

File: api/api_test.go (hunk 1 above)
Comment (nharper) [reply api/api_test.go:38]: I think that `server.removable` is correct for these, as they are error codes for the Removable API endpoint, not the Remove API endpoint. (E.g. there's both a `server.removable.protected` and a `server.remove.protected` error code, depending on whether the request was made to /api/v2/remove or /api/v2/removable.)

**Pull request merged**